### PR TITLE
Rename, fix and add tests for IndexPath position comparison

### PR DESCRIPTION
**Pull request by elland**
- Addresses issue #1

### CODE REVIEW

**Comment by elland** [comment]
## Waiting on https://github.com/bakkenbaeck/SweetFoundation/pull/4

File: Demo.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -300,6 +306,8 @@
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
 				PRODUCT_BUNDLE_IDENTIFIER = "com.sample.$(PRODUCT_NAME:rfc1034identifier)";
 				PRODUCT_NAME = "$(TARGET_NAME)";
+				SWIFT_OPTIMIZATION_LEVEL = "-Owholemodule";
+				SWIFT_VERSION = 2.3;
```
Comment (3lvis) [inline Demo.xcodeproj/project.pbxproj]: 💀

File: Sources/NSIndexPath+Sweetness.swift [hunk 2]
```diff
@@ -2,21 +2,28 @@ import UIKit
...
+        case Before
+        case Ahead
     }
 
-    public func compareDirection(indexPath: NSIndexPath) -> Direction {
-        let current = self.row * self.section
-        let coming = indexPath.row * indexPath.section
-
-        if current == coming {
+    private func comparePosition(ownRow ownRow: Int, otherRow: Int) -> Direction {
```
Comment (3lvis) [inline Sources/NSIndexPath+Sweetness.swift]: `ownRow` sounds kind of weird. I think is fine just by not having a parameter name here.

File: Sources/NSIndexPath+Sweetness.swift [hunk 3]
```diff
@@ -2,21 +2,28 @@ import UIKit
 
 public extension NSIndexPath {
     public enum Direction {
-        case Forward
-        case Backward
         case Same
+        case Before
+        case Ahead
```
Comment (3lvis) [inline Sources/NSIndexPath+Sweetness.swift:7]: This reads much better :clap:

File: Demo.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -147,11 +147,14 @@
 			attributes = {
 				CLASSPREFIX = "";
 				LastSwiftUpdateCheck = 0700;
-				LastUpgradeCheck = 0700;
+				LastUpgradeCheck = 0800;
```
Comment (3lvis) [inline Demo.xcodeproj/project.pbxproj:150]: Hello Mr. Bleeding Edge. How are beta bugs treating you?

Comment (elland) [inline Demo.xcodeproj/project.pbxproj:150]: Just as buggy as any Xcode, really...

File: Sources/NSIndexPath+Sweetness.swift (hunk 3 above)
Comment (elland) [inline Sources/NSIndexPath+Sweetness.swift:7]: 🎸 📚

**Comment by 3lvis** [comment]
tumblr_o1ea1upbl21rpeqpko1_500

**Comment by elland** [comment]
And now fix #1 💃

**Pull request merged**